external/CMakeLists.txt

```diff
@@ -51,16 +51,27 @@ set(OLP_SDK_CPP_RAPIDJSON_TAG "master")
 set(OLP_SDK_CPP_BOOST_URL "https://dl.bintray.com/boostorg/release/1.69.0/source/boost_1_69_0.tar.gz")
 
 # Add external projects
-add_subdirectory(googletest)
+find_package(GTest QUIET)
+if (NOT GTest_FOUND)
+    add_subdirectory(googletest)
+endif()
+
+find_package(RapidJSON QUIET)
+if(NOT RapidJSON_FOUND)
+    add_subdirectory(rapidjson)
+    set(RapidJSON_DIR ${EXTERNAL_RapidJSON_DIR} PARENT_SCOPE)
+endif()
 
-add_subdirectory(rapidjson)
-set(RapidJSON_DIR ${EXTERNAL_RapidJSON_DIR} PARENT_SCOPE)
+find_package(leveldb QUIET)
```

Comment:
Is `snappy` handled this way?

Reply:
snappy is downloaded as a part of leveldb build